### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Pavlo000**

### CODE REVIEW

File: src/arrayMethodJoin.js [hunk 1]
```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let mySeparator = separator;
+    let join = '';
+
+    if (separator === undefined) {
+      mySeparator = ',';
+    }
```
Comment (jstmpelowycz) [inline src/arrayMethodJoin.js:13]: to avoid this, you could give `separator` a default value

File: src/arrayMethodJoin.js [hunk 2]
```diff
@@ -5,7 +5,25 @@
...
+
+    for (let index = 0; index < this.length; index++) {
+      if (this.length - index > 1) {
+        join += `${
+          this[index] === null
+          || this[index] === undefined ? '' : this[index]}${mySeparator
+        }`;
+      } else {
+        join += this[index] === undefined ? '' : this[index];
+      }
```
Comment (jstmpelowycz) [inline src/arrayMethodJoin.js:23]: this structure is too complicated, I'm sure you could get it simpler

**Review by jstmpelowycz — CHANGES_REQUESTED** [review]

**Comment by Pavlo000** [comment]
I don’t know how to do it easier, maybe you can give some hint?

ср, 8 черв. 2022 р. о 17:10 Dmytro Pelovych ***@***.***> пише:

File: src/arrayMethodJoin.js [hunk 3]
```diff
@@ -4,8 +4,20 @@
...
+    let join = '';
+
+    for (const item of this) {
+      if (item !== undefined && item !== null) {
+        join += item;
+      }
+
+      if (this.indexOf(item) !== this.length - 1) {
+        join += separator;
+      }
```
Comment (jstmpelowycz) [inline src/arrayMethodJoin.js:17]: suggestion: I'd rather use for-with-indexes loop to avoid using `indexOf`, but still – up to you

**Review by jstmpelowycz — APPROVED** [review]

**Pull request closed without merging**